### PR TITLE
Bug 1294746 - Disable more autoclassify buttons when not logged in

### PR DESCRIPTION
**Pull request by KWierso**
So, this PR as-is:
1) Disables the "ignore others" button and changes the title for the button when not signed in.
2) Changes the titles for the line-specific "save" buttons and a select dropdown, and attempts to disable both when not signed in.

But those attempts to disable them don't appear to work, and I don't know why, since "loggedIn()" is working just above the ng-disabled's as part of the titles... @jgraham ?

---
This change is [Reviewable](https://reviewable.io/reviews/mozilla/treeherder/1784)

### CODE REVIEW

File: ui/plugins/auto_classification/main.html [hunk 1]
```diff
@@ -39,11 +39,13 @@
           <select
              ng-if="line.type === 'structured' && line.updateText.length > 1"
              ng-model="line.updateSelectOption"
+             title="{{loggedIn() ? '' : 'Must be logged in'}}"
+             ng-disabled="!loggedIn"
```
Comment (wlach) [inline ui/plugins/auto_classification/main.html]: You need to call loggedIn() as a function, not just get the value of the variable (which is always true).

**Comment by KWierso** [comment]
I also had the logic flipped around, which explains why that didn't work for me earlier. Fixed, thanks!

File: ui/plugins/auto_classification/controller.js [hunk 2]
```diff
@@ -976,7 +976,7 @@ treeherder.controller('ClassificationPluginCtrl', [
         };
 
         $scope.canIgnore = function() {
-            return $scope.pendingLines().length > 0;
+            return $scope.pendingLines().length > 0 && $scope.loggedIn();
```
Comment (wlach) [inline ui/plugins/auto_classification/controller.js]: Actually now that I look at it, this change isn't necessary. It's probably better to just restrict this at the html/template level, as you are.

File: ui/plugins/auto_classification/main.html [hunk 3]
```diff
@@ -39,11 +39,13 @@
...
              ng-if="line.type === 'structured' && line.updateText.length > 1"
              ng-model="line.updateSelectOption"
+             title="{{loggedIn() ? '' : 'Must be logged in'}}"
+             ng-disabled="!loggedIn()"
              ng-options="value for (key, value) in line.updateText"
              ></select><br>
           <button class="btn btn-xs btn-default"
-                  title="Select best match and click here"
-                  ng-disabled="!line.canSave"
+                  title="{{loggedIn() ? 'Select best match and click here' : 'Must be logged in'}}"
```
Comment (wlach) [inline ui/plugins/auto_classification/main.html]: I guess we might want to say something if the canSave is false, but the user is not logged in ("No lines to save"?).

**Comment by KWierso** [comment]
@wlach 
How about this? I'm not super thrilled with the length of some of these lines, and a bunch of these strings are now duplicated. Should I be moving them somewhere (values.js, into the controller) so they can be re-used and referred to in shorter lines?

File: ui/plugins/auto_classification/main.html [hunk 4]
```diff
@@ -4,8 +4,8 @@
   <div class="failure-lines-actions" ng-if="status() === 'pending'">
     <button ng-if="failureLines"
             class="btn btn-primary btn-sm"
-            title="Ignore items with no autoclassification and nothing explicitly selected"
-            ng-disabled="!canIgnore()"
+            title="{{loggedIn() ? 'Ignore items with no autoclassification and nothing explicitly selected': 'Must be logged in to change classifications'}}"
```
Comment (wlach) [inline ui/plugins/auto_classification/main.html]: I would shorten this to just "Not logged in".

File: ui/plugins/auto_classification/main.html [hunk 5]
```diff
@@ -39,11 +39,13 @@
           <select
              ng-if="line.type === 'structured' && line.updateText.length > 1"
              ng-model="line.updateSelectOption"
+             title="{{loggedIn() ? '' : 'Must be logged in to change classifications'}}"
```
Comment (wlach) [inline ui/plugins/auto_classification/main.html]: I would change this to 'Update selected classifications' or 'Not logged in'

File: ui/plugins/auto_classification/main.html [hunk 6]
```diff
@@ -39,11 +39,13 @@
...
+             ng-disabled="!loggedIn()"
              ng-options="value for (key, value) in line.updateText"
              ></select><br>
           <button class="btn btn-xs btn-default"
-                  title="Select best match and click here"
-                  ng-disabled="!line.canSave"
+                  title="{{loggedIn() ? 'Select best match and click here' : 'Must be logged in to change classifications\n'}}{{line.canSave ? '' : 'No changes to save'}}"
```
Comment (wlach) [inline ui/plugins/auto_classification/main.html]: So you'll be printing double here. I would think that the logged in status would be most significant here, so let's print that first. Something like this should do the trick:

```
title="{{loggedIn() ? (line.canSave ? 'Select best match' : 'No changes to save') : 'Not logged in'}}
```

**Comment by wlach** [comment]
For maximum reuse, we could move this "Disable button if not logged in" behaviour into an angular directive (or component if we wanted to upgrade to Angular 1.5, which we probably should), but that might be best done in a followup. 

For here, I would simplify the string to just "Not logged in" here: I don't think there's any need to call out what specifically the user can't do in that case in the UI.
